keyboards/handwired/ferris/matrix.c

```diff
@@ -0,0 +1,261 @@
+/*
```

Comment:
Can this file be converted over to use `CUSTOM_MATRIX = lite`?
If it's possible, you'll only need to specify `matrix_init_custom()` and `matrix_scan_custom()`, with the majority of the other common things like `matrix_is_on()` etc. implemented for you.

Reply:
Thanks a lot for the feedback. I spent some time trying to do this, but couldn't get to a state where any keypress is recognized with CUSTOM_MATRIX=lite, so ended up keeping the boilerplate that I know how to make work. I hope that's OK.

Reply:
Nevermind. I found the bug in my code. I wasn't returning the correct status from matrix_scan_custom. Now works, so I added a commit doing this. It's a really nice cleanup, thanks for teaching me about it :smile: